Yoast/javascript
# P2 671 analysis for job posting not always shown

## Summary

Fixes missing analysis for job posting.


This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Improves the sidebar for Yoast structured data blocks.

## Relevant technical choices:

* the generic Block will now check if any of the current block's parents are yoast/job-posting or yoast/recipe. if so, show that parent's sidebar instead of the current block's sidebar.
* known issue: only the analysis is shown; we need to pass the parent block's *options* too if we want further details in the sidebar.

## Test instructions

This PR can be tested by following these steps:

* add a job block
* navigate up/down with cursor keys until the entire job posting is selected. notice the analysis in the sidebar.
* navigate up/down to any of the job posting's innerblocks.
* notice the analysis is still in the sidebar, *but the other sidebar fields are now missing.*

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes #
